urllib3/connection.py

```diff
@@ -204,13 +205,21 @@ class HTTPSConnection(HTTPConnection):
     default_port = port_by_scheme['https']
 
     def __init__(self, host, port=None, key_file=None, cert_file=None,
-                 strict=None, timeout=socket._GLOBAL_DEFAULT_TIMEOUT, **kw):
+                 strict=None, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
+                 ssl_context=None, ssl_version=None, **kw):
```

Comment:
I think I did this on the first attempt at this PR. The logic is preserving backwards compatibility because it already accepts `ssl_version`. Most people using `ssl_version` probably will not want to switch to creating a context object.

Would it be better for them to be mutually exclusive?